### ISSUE TITLE
Why are datapoints (dps) empty for a given timerange?

### ISSUE THREAD

**Issue by buddyp450** [OpenTSDB/opentsdb#672]
If I invoke a normal opentsdb query for a time range like:

```
curl 'http://localhost:4241/api/query?start=2015/10/01end=2015/11/30&m=sum:1d-sum:mymetric{tag1=*,tag2=value2,tag3=value3}'
```

How is it possible that I can get results that do not have datapoints? Isn't the point of the time query to return values that only have datapoints in them?

**Comment by manolama** [comment]
Yeah this is a funny situation we need to clean up. What happens is that TSD will request a row of data before and after your start and end times (with a bit more possibly when downsampling is enabled). That's because data is stored in HBase on an hourly basis per row. We fetch a bit extra so that if you request a rate, we can return the earliest rate within the range you specified. 

So what happens when you see an empty set is that the TSD found some data before or after your endtime but filtered it all out. We need to do a better job of not returning anything if there isn't any data falling within the time range.

**Comment by zjryan** [comment]
and at which version will this bug fixed?

**Comment by manolama** [comment]
We'll get this fixed for 2.4, sorry for the wait.

**Comment by pheoby** [comment]
@manolama When will the v2.4.0 be released?

**Comment by MrKingHH** [comment]
Has the bug been fixed? @manolama

**Comment by mcauto** [comment]
Same here. 

also unix time.

**Comment by johann8384** [comment]
I can look into getting this fix in if there isn't one already, what do you mean @mcauto by "also Unix time."

**Comment by mcauto** [comment]
@johann8384 

I collect metrics every minutes.

Datapoints exists. however, always return empty dps.

**empty**
```
http://localhost:4242/api/query?start=1598490000&end=1598490059&m=...
```

**one point**
```
http://localhost:4242/api/query?start=1m-ago&end=1598490059&m=...
```

What is the difference?

**Comment by balboah** [comment]
seems this bug still exists in 2.4.1
